### DISCUSSION TITLE
Next.js deploy "next start"?

### DISCUSSION THREAD

**Discussion by bluetidepro** [vercel/vercel#4437] [Help]
When I deploy my Next.js app, I see it runs `npm run build`, but the logs never show `npm start`, `npm run start`, or even `next start` - to start the server up after the build. Am I missing something? Do I need to set that up in the settings, or is it auto ran?

I'm having troubles getting my build to run when deployed, but it works fine locally when I run `npm i`, `npm run build`, then `next start` (or `npm start`, or `npm run start`).

Here are the commands in my `package.json`:

```
  "scripts": {
    "analyze": "node -v && cross-env-shell NODE_OPTIONS=\"-r esm\" ANALYZE=true next build",
    "build": "node -v && cross-env NODE_OPTIONS=\"-r esm\" NODE_ENV=production next build",
    "dev": "node -v && cross-env NODE_OPTIONS=\"-r esm\" next",
    "lint": "eslint '**/*.js'",
    "pretty": "prettier --single-quote --trailing-comma all --write package.json '**/*.js'",
    "start": "node -v && cross-env NODE_OPTIONS=\"-r esm\" NODE_ENV=production next start"
  },
```

Not sure if it has to do with needing the `esm` option or not. But those are the commands that work locally.

**Comment by paulogdm** [comment]
Vercel is a serverless platform with a focus on static-first deployments. Therefore, "starting servers" is simply not supported. You can take a look at the following resources for examples:

1. https://vercel.com/guides
2. https://github.com/zeit/now/tree/master/examples

**Reply by bluetidepro** [reply]
@paulogdm I guess I’m still a bit confused. Why does my app work locally with “vercel dev” but not when I deploy it to production. It seems to mainly be something with the “esm” importing I use, I think. Something about the imports. I will have to post the exact error. I just am so lost how it works locally, but then fails once I deploy. Haha
